compiler/src/iree/compiler/API/Internal/Embed.cpp

```diff
@@ -355,15 +452,34 @@ Error *Output::openFile(const char *filePath) {
 }
 
 Error *Output::openFD(int fd) {
+  type = Type::File;
   description = "fd-";
   description.append(std::to_string(fd));
   outputFile = std::make_unique<llvm::ToolOutputFile>(description, fd);
   // Don't try to delete, etc.
   outputFile->keep();
   outputStream = &outputFile->os();
+  this->backingFileDescriptor = fd;
```

Comment:
why the this->s? (don't seem to be needed in most of the new lines? I see this-> and immediately think template metaprogramming type ambiguity hacks :)

Reply:
I just don't like it when I have a bunch of instance and local variables that are named similarly and the job of a function is to set the things at the instance level. Beyond a certain threshold, I just add `this->`. Call it my stress reaction when not using a code style that syntactically differentiates between members and variables :)